src/software/sensor_fusion/sensor_fusion_test.cpp

```diff
@@ -294,6 +341,93 @@ class SensorFusionTest : public ::testing::Test
+    *(sensor_msg.mutable_ssl_vision_msg()) = *ssl_wrapper_packet;
+    *(sensor_msg.add_robot_status_msgs())  = *robot_status_msg_wheel_motor_hot;
+    sensor_fusion.processSensorProto(sensor_msg);
+
+    std::optional<Robot> robot =
+        sensor_fusion.getWorld().value().friendlyTeam().getRobotById(2);
+    ASSERT_TRUE(robot);
+    std::set<RobotCapability> robot_unavailable_capabilities =
+        robot.value().getUnavailableCapabilities();
+    EXPECT_EQ(1, robot_unavailable_capabilities.size());
+    bool is_movement_disabled =
+        robot_unavailable_capabilities.find(RobotCapability::Move) !=
+        robot_unavailable_capabilities.end();
+    ASSERT_TRUE(is_movement_disabled);
+}
+
+TEST_F(SensorFusionTest,
+       test_making_chip_and_kick_robot_capabilities_unavailable_from_error_code)
+{
+    SensorProto sensor_msg;
+    auto ssl_wrapper_packet =
+        createSSLWrapperPacket(std::move(geom_data), initDetectionFrame());
+    *(sensor_msg.mutable_ssl_vision_msg()) = *ssl_wrapper_packet;
+    *(sensor_msg.add_robot_status_msgs())  = *robot_status_msg_low_cap;
+    sensor_fusion.processSensorProto(sensor_msg);
+
+    Team friendly_team         = sensor_fusion.getWorld().value().friendlyTeam();
+    std::optional<Robot> robot = friendly_team.getRobotById(2);
+    ASSERT_TRUE(robot);
+    std::set<RobotCapability> robot_unavailable_capabilities =
+        robot.value().getUnavailableCapabilities();
+    EXPECT_EQ(2, robot_unavailable_capabilities.size());
+
+    bool is_kick_disabled = robot_unavailable_capabilities.find(RobotCapability::Kick) !=
+                            robot_unavailable_capabilities.end();
+    ASSERT_TRUE(is_kick_disabled);
+
+    bool is_chip_disabled = robot_unavailable_capabilities.find(RobotCapability::Chip) !=
+                            robot_unavailable_capabilities.end();
+    ASSERT_TRUE(is_chip_disabled);
+}
+
+TEST_F(SensorFusionTest,
+       test_making_dribble_robot_capabilities_unavailable_from_error_code)
+{
+    SensorProto sensor_msg;
+    auto ssl_wrapper_packet =
+        createSSLWrapperPacket(std::move(geom_data), initDetectionFrame());
+    *(sensor_msg.mutable_ssl_vision_msg()) = *ssl_wrapper_packet;
+    *(sensor_msg.add_robot_status_msgs())  = *robot_status_msg_dribble_motor_hot;
+    sensor_fusion.processSensorProto(sensor_msg);
+
+    Team friendly_team         = sensor_fusion.getWorld().value().friendlyTeam();
+    std::optional<Robot> robot = friendly_team.getRobotById(2);
+    ASSERT_TRUE(robot);
+    std::set<RobotCapability> robot_unavailable_capabilities =
+        robot.value().getUnavailableCapabilities();
+    EXPECT_EQ(1, robot_unavailable_capabilities.size());
+
+    bool is_kick_disabled =
+        robot_unavailable_capabilities.find(RobotCapability::Dribble) !=
+        robot_unavailable_capabilities.end();
+    ASSERT_TRUE(is_kick_disabled);
+}
+
+TEST_F(SensorFusionTest, test_emptying_robot_unavailable_capabilities_from_error_code)
```

Comment:
can you add a test where you update the robot capabilities multiple times with different capabilities and assert you get the correct one at the end

Reply:
Added!